src/client.rs

```diff
@@ -98,14 +119,14 @@ pub extern "C" fn rustls_client_config_builder_new_with_safe_defaults(
 /// `rustls_tls_version` for convenience.
 ///
 /// `versions` will only be used during the call and the application retains
-/// ownership. `len` is the number of consecutive `ui16` pointed to by `versions`.
+/// ownership. `len` is the number of consecutive `u16` pointed to by `versions`.
 #[no_mangle]
-pub extern "C" fn rustls_client_config_builder_new(
+pub extern "C" fn rustls_client_config_builder_new_fancy(
```

Comment:
It seems weird that some options are set using `rustls_client_config_builder_*` functions but some others can only be set in this function's arguments. If you had something like `rustls_client_config_builder_cipher_suites()` and `rustls_client_config_builder_protocol_versions()`, could this function be deleted?

Reply:
I've merged main, and tested (locally) a client that only uses one ciphersuite and one TLS version; it doesn't build any smaller. I need to experiment a bit more to see if I can make this work as expected.

Reply:
I haven't found time to figure out how to make the dead code elimination work as expected. I need to try compiling rustls itself with default ciphersuites and with a narrowed set of ciphersuites to see if it works upstream. Meanwhile, though, I'd like to proceed with this API, since I suspect it will be required if we ever want dead code elimination to work.